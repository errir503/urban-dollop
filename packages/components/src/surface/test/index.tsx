/**
 * External dependencies
 */
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import { Surface } from '../index';

describe( 'props', () => {
	test( 'should render correctly', () => {
		const { container } = render( <Surface>Surface</Surface> );
		expect( container ).toMatchSnapshot();
	} );

	test( 'should render variants', () => {
		const view = render( <Surface>Surface</Surface> );
		const { container } = render(
			<Surface variant="secondary">Surface</Surface>
		);
		expect( container ).toMatchDiffSnapshot( view.container );
	} );

	test( 'should render borderLeft', () => {
		const view = render( <Surface>Surface</Surface> );
		const { container } = render( <Surface borderLeft>Surface</Surface> );
		expect( container ).toMatchDiffSnapshot( view.container );
	} );

	test( 'should render borderRight', () => {
		const view = render( <Surface>Surface</Surface> );
		const { container } = render( <Surface borderRight>Surface</Surface> );
		expect( container ).toMatchDiffSnapshot( view.container );
	} );

	test( 'should render borderTop', () => {
		const view = render( <Surface>Surface</Surface> );
		const { container } = render( <Surface borderTop>Surface</Surface> );
		expect( container ).toMatchDiffSnapshot( view.container );
	} );

	test( 'should render borderBottom', () => {
		const view = render( <Surface>Surface</Surface> );
		const { container } = render( <Surface borderBottom>Surface</Surface> );
		expect( container ).toMatchDiffSnapshot( view.container );
	} );
} );
