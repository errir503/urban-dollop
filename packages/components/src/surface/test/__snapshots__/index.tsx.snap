// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props should render borderBottom 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -1,8 +1,8 @@
  <div>
    <div
-     class="components-surface css-pt58n0-View-Surface-getBorders-primary e19lxcc00"
+     class="components-surface css-soe81k-View-Surface-getBorders-primary e19lxcc00"
      data-wp-c16t="true"
      data-wp-component="Surface"
    >
      Surface
    </div>
`;

exports[`props should render borderLeft 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -1,8 +1,8 @@
  <div>
    <div
-     class="components-surface css-1vckp4o-View-Surface-getBorders-primary e19lxcc00"
+     class="components-surface css-soe81k-View-Surface-getBorders-primary e19lxcc00"
      data-wp-c16t="true"
      data-wp-component="Surface"
    >
      Surface
    </div>
`;

exports[`props should render borderRight 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -1,8 +1,8 @@
  <div>
    <div
-     class="components-surface css-sw9dzi-View-Surface-getBorders-primary e19lxcc00"
+     class="components-surface css-soe81k-View-Surface-getBorders-primary e19lxcc00"
      data-wp-c16t="true"
      data-wp-component="Surface"
    >
      Surface
    </div>
`;

exports[`props should render borderTop 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -1,8 +1,8 @@
  <div>
    <div
-     class="components-surface css-123k66h-View-Surface-getBorders-primary e19lxcc00"
+     class="components-surface css-soe81k-View-Surface-getBorders-primary e19lxcc00"
      data-wp-c16t="true"
      data-wp-component="Surface"
    >
      Surface
    </div>
`;

exports[`props should render correctly 1`] = `
.emotion-0 {
  background-color: #fff;
  color: #1e1e1e;
  position: relative;
}

<div>
  <div
    class="components-surface emotion-0 emotion-1"
    data-wp-c16t="true"
    data-wp-component="Surface"
  >
    Surface
  </div>
</div>
`;

exports[`props should render variants 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -1,8 +1,8 @@
  <div>
    <div
-     class="components-surface css-1m2pafr-View-Surface-getBorders-secondary e19lxcc00"
+     class="components-surface css-soe81k-View-Surface-getBorders-primary e19lxcc00"
      data-wp-c16t="true"
      data-wp-component="Surface"
    >
      Surface
    </div>
`;
